src/main/java/nl/ai42/controllers/LoginAndRegisterController.java

```diff
@@ -17,84 +19,120 @@ public class LoginAndRegisterController {
-                loginUsernameTextField.setStyle(errorStyle);
-            } else if (loginPasswordPasswordField.getText().isBlank()) {
-                loginPasswordPasswordField.setStyle(errorStyle);
+        this.resetFields();
+
+        if (login_username_text_field.getText().isBlank() || login_password_password_field.getText().isBlank()) {
+            invalid_login_credentials.setText("Not all required fields are filled in.");
+            invalid_login_credentials.setStyle(errorMessage);
+            invalid_signup_credentials.setText("");
+
+            if (login_username_text_field.getText().isBlank()) {
+                login_username_text_field.setStyle(errorStyle);
+            }
+            if (login_password_password_field.getText().isBlank()) {
+                login_password_password_field.setStyle(errorStyle);
             }
         } else {
-            invalidLoginCredentials.setText("Login Successful!");
-            invalidLoginCredentials.setStyle(successMessage);
-            loginUsernameTextField.setStyle(successStyle);
-            loginPasswordPasswordField.setStyle(successStyle);
-            invalidSignupCredentials.setText("");
+            invalid_login_credentials.setText("Login Successful!");
+            invalid_login_credentials.setStyle(successMessage);
+            login_username_text_field.setStyle(successStyle);
+            login_password_password_field.setStyle(successStyle);
+            invalid_signup_credentials.setText("");
         }
     }
 
     @FXML
     protected void onSignUpButtonClick() {
+        this.resetFields();
+
+        boolean error = false;
 
-        if (signUpUsernameTextField.getText().isBlank() || signUpEmailTextField.getText().isBlank() ||
-                signUpPasswordPasswordField.getText().isBlank() || signUpRepeatPasswordPasswordField.getText().isBlank()
+        if (sign_up_username_text_field.getText().isBlank() || sign_up_email_text_field.getText().isBlank() ||
+                sign_up_password_password_field.getText().isBlank() || sign_up_repeat_password_password_field.getText().isBlank()
         ) {
-            invalidSignupCredentials.setText("Please fill in all fields!");
-            invalidSignupCredentials.setStyle(errorMessage);
-            invalidLoginCredentials.setText("");
+            error = true;
+            invalid_signup_credentials.setText("Not all required fields are filled in.");
+            invalid_signup_credentials.setStyle(errorMessage);
+
+            if (sign_up_username_text_field.getText().isBlank()) sign_up_username_text_field.setStyle(errorStyle);
+            if (sign_up_email_text_field.getText().isBlank()) sign_up_email_text_field.setStyle(errorStyle);
+            if (sign_up_password_password_field.getText().isBlank()) sign_up_password_password_field.setStyle(errorStyle);
+            if (sign_up_repeat_password_password_field.getText().isBlank()) sign_up_repeat_password_password_field.setStyle(errorStyle);
```

Comment:
format it below the if-statement not right after its cleaner that way, lets do that everywhere

Reply:
Committed